### ISSUE TITLE
[bug report] Cancel uploading file

### ISSUE THREAD

**Issue by OnlyBelter** [bleenco/ngx-uploader#345]
When I cancel a uploading file, the `f.progress.data.percentage` value will still goes up to 100.

**Comment by OnlyBelter** [comment]
I add this to my `onUploadOutput` funciton:

```
if (output.type === 'cancelled') {
      console.log('you cancelled me');
      const index = this.files.findIndex(file => typeof output.file !== 'undefined' && file.id === output.file.id);
      this.files[index].progress.data.percentage = 0;
    }
```

But, it will keep update to 100, then I don't know how to stop this progress.

**Comment by OnlyBelter** [comment]
Please reference here, https://github.com/AckerApple/angular-file/blob/development/src/file-upload/FileUploader.class.ts#L204, If anyone can understand all the code. I can understand a little, cannot give a pull request for this problem.

**Comment by OnlyBelter** [comment]
Fixed in v4.0, thanks a lot!
